platform/lib/tools/growReferenceChecker.js

```diff
@@ -25,13 +25,29 @@ const fs = require('fs');
 // Where to look for existing documents
 const POD_BASE_PATH = path.join(__dirname, '../../../pages/');
 // Which documents to check for broken references
-const PAGES_SRC = POD_BASE_PATH + 'content/amp-dev/documentation/**/*.{md,html}';
+const PAGES_SRC = POD_BASE_PATH + 'content/amp-dev/**/*.{md,html}';
 // The location to search for documents in
 const PAGES_BASE_PATH = POD_BASE_PATH + 'content/amp-dev';
-// The pattern used by Grow to make up references
-const REFERENCE_PATTERN = /\[[^\]]+\]\(([^:\)\{?#]+)(?:[?#][^\)]*)?\)|g.doc\('(.*?)'/g;
-// Contains manual hints for double filenames etc.
+// The pattern to find links in markdown and html
+// It also matches source code blocks to skip these
+const REFERENCE_PATTERN = new RegExp(
+    // skip sourcecode block in markdown:
+    /^```[\s\S]*?```/.source
```

Comment:
won't this fail for "```html"?

Reply:
No, the "[\s\S]" will match any character until the ending ``` is found.

Reply:
ah cool -  thanks